cmd/controller-manager/app/controllermanager.go

```diff
@@ -604,8 +606,9 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 		return
 	}
 
+	karmadaClient := karmadaclientset.NewForConfigOrDie(restConfig)
 	controlPlaneInformerManager := genericmanager.NewSingleClusterInformerManager(dynamicClientSet, 0, stopChan)
-
+	factory := informerfactory.NewSharedInformerFactory(karmadaClient, 0)
```

Comment:
Will there be any overlap between this `factory` and the `controlPlaneInformerManager` on above?
If the same resource is registered(watched) by two informers, memory usage will be doubled.

Reply:
> If the same resource is registered(watched) by two informers, memory usage will be doubled.

Currently, only the `ResourceBinding` resource is registered in the `factory`.

Reply:
I want to use `structured RB` and avoid doing type conversions every time. 😂😂😂

Reply:
> I want to use structured RB and avoid doing type conversions every time.

I know, the typed informer is more convenient to use and also saves CPU costs by avoiding transformation.

However, memory usage is also a factor that must be considered, otherwise, it will bring scalability issues.

> Currently, only the ResourceBinding resource is registered in the factory.

Remember that, the `controller-runtime` also manages a cache which also includes `ResourceBinding`, can we retrieve(list) object from it?

Reply:
> the controller-runtime also manages a cache which also includes ResourceBinding, can we retrieve(list) object from it?



Ok, i can try it.